#! /usr/bin/env python

PACKAGE='youbot_arm_ik'
try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

method_enum = gen.enum([
  gen.const("closest_ik", int_t, 0, "closest_ik"),
  gen.const("preferred_pitch_ik",   int_t, 1, "preferred_pitch_ik, need to set pitch value"),
  gen.const("preferred_type_ik",    int_t, 2, "preferred_type_ik, need to set arm_to_front, arm_bended_up and gripper_downwards"),
  gen.const("fully_constrained_ik", int_t, 3, "fully_constrained_ik, need to set id and pitch")
],"Ik method")

gen.add("method",    int_t, 0, "Ik method", 0, 0, 3, edit_method=method_enum)
gen.add("timeout",   double_t, 0, "timeout", 20, 0, 120)
gen.add("duration",  double_t, 0, "duration", 5, 0, 60)
gen.add("x",    double_t, 0, "x axis", 0.5, -1.0, 1.0)
gen.add("y",    double_t, 0, "y axis", 0.5, -0.5, 0.5)
gen.add("z",    double_t, 0, "z axis", 0.5, -0.5, 1.0)
gen.add("roll",    double_t, 0, "roll", 0.0, -3.14, 3.14)
gen.add("pitch",    double_t, 0, "pitch", 0.0, -3.14, 3.14)
gen.add("yaw",    double_t, 0, "yaw", 0.0, -3.14, 3.14)
gen.add("preferred_pitch",double_t, 0, "pitch for preferred_pitch_ik", 0, -3.14, 3.14)
gen.add("id",   int_t, 0, "id", 0, 0, 10)
gen.add("arm_to_front", bool_t, 0, "arm_to_front", False)
gen.add("arm_bended_up", bool_t, 0, "arm_bended_up", False)
gen.add("gripper_downwards", bool_t, 0, "gripper_downwards", False)
gen.add("move_arm", bool_t, 0, "move_arm", False)

exit(gen.generate(PACKAGE, "youbot_arm_ik", "Controller"))
